Signal migration follow-ups

devversion opened the pull request:
See individual commits

devversion committed:
test: improve typescript version coverage for signal input migration

This ensures the migration works for these TypeScript versions. The
migration is very sensitive to the TS version and its internals; so it
makes sense to test all of these.

devversion committed:
test: add test to verify extending tsconfig works in signal input migration

Related to https://github.com/angular/angular/issues/59348

crisbeto commented on packages/core/schematics/migrations/signal-migration/test/ts-versions/index.bzl:
```diff
@@ -1,7 +1,11 @@
-    "typescript-5.5.4",
-    "typescript-5.5.3",
     "typescript-5.5.2",
+    "typescript-5.5.3",
```
Should we just do one version per minor? e.g. 5.5, 5.6 and 5.7? AFAIK there shouldn't be huge behavior differences between them.

devversion replied:
I think you're right, but I also don't think it hurts having this increased coverage. WDYT?

crisbeto replied:
I don't feel strongly about it, but it'll slow down the CI a bit and might make it more noisy 🤷‍♂️

devversion replied:
It shouldn't run very often, but yeah. Let's just keep an eye on it and we can reduce soon anyway IIUC

crisbeto commented on packages/core/schematics/test/signal_input_migration_spec.ts:
```diff
@@ -0,0 +1,97 @@
+    expect(content).toContain('readonly name = input.required<string>()');
+  });
+
+  fit('should work when extending tsconfig from node_modules', async () => {
```
Leftover `fit`

devversion replied:
Fixed

devversion committed:
fixup! test: add test to verify extending tsconfig works in signal input migration

ds

devversion committed:
fix(migrations): incorrect stats when migrating queries with best effort mode

We previously did count forcibly ignored queries as incompatible. This
resulted in incorrect migration stats that are printed upon migration
completion.

See: #58657

kirjs commented:
This PR was merged into the repository by commit 8496fb16bc081a19cd55b26dd7bf86d96dd18114.

The changes were merged into the following branches: main, 19.0.x, 19.1.x
